### ISSUE TITLE
Dandi Upload currently lacks an option to add custom keys to dandi file names 

### ISSUE THREAD

**Issue by Ahad-Allen** [dandi/dandi-cli#1235]
Currently, it appears that dandi organize/upload are designed to handle file names with the following format sub-<subject_id>[_key-<value>][_mod1+mod2+...].nwb. This works well in general, but means that adding other specifiers to help identify files can be a bit tricky. For example, the Allen Institute has an experiment id associated with each plane of ophys sessions that is difficult to add to filepaths due to not being a recognized key value pair.
I am not fully informed about the standards dandi follows for file naming conventions, but I feel that allowing users to include custom key value pairs in their file names would be a useful addition if it does not cause any conflicts with the standards dandi follows.

**Comment by yarikoptic** [comment]
DANDI accepts datasets in BIDS format and since BIDS was not (WiP) fully ready to cover animal neurophys data, we came up with  `organize` command which formalizes into BIDS-like hierarchy but with some divergences. Indeed we should document better (hence added a label) on which metadata entities already participate in filename construction. ATM you can find them in https://github.com/dandi/dandi-cli/blob/HEAD/dandi/consts.py#L177 . Overall we would be open to add probably any entity already covered by BIDS: https://bids-specification.readthedocs.io/en/stable/appendices/entities.html . Looking more specifically at https://bids-specification.readthedocs.io/en/stable/04-modality-specific-files/10-microscopy.html - is any one in particular is best aligned (`sample`, `acq`?) with the metadata (experiment id could even be session id for me) in this case?

**Comment by Ahad-Allen** [comment]
Thanks for the response @yarikoptic. Going over that documentation, I think the acq label is probably the best option for adding custom information in this case. It seems to be intended for custom labels designed to distinguish between similar sessions with experimental differences anyway.

**Comment by yarikoptic** [comment]
yes. And it is quite flexible and not anyhow regulated.  But that then also limits possibility for automation. E.g. in your case I would expect `dandi organize` to (ab)use non-informative `_id-` field to distinguish two files which are otherwise the same in terms of metadata.  I wonder if we should add some "plug-in" functionality so you could add to your `dandi organize` invocation and it would decide on what custom `key`s to add based on the files in question.

**Comment by Ahad-Allen** [comment]
I think that could be quite useful for automation, but may lose some of the flexibility provided by the acq label. Would being able to manually specify additional custom keys in the dandi organize function be possible? Also, what would be the time frame for adding such a feature or the acq label? Thank you so much for your help @yarikoptic

**Comment by yarikoptic** [comment]
Well, there is already nothing which forbids you from
- running `dandi organize` to organize using `_id-ID` to disambiguate
- rename all the files replacing non-informative `_id-ID` with `_acq-custom1` and alike. 
since AFAIK we are not validating really those keys ATM in the layout and `dandi validate` (and thus `dandi upload`) should not complain.

Do you see/have some python code which, e.g. given an nwb file, could provide a decision on the value of `_acq-` to use for it?
